pkg/transforms/policy.go

```diff
@@ -37,6 +41,106 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
 	if okns && okpp {
 		node.Properties["_parentPolicy"] = pnamespace + "/" + ppolicy
 	}
+
+	return &PolicyResource{node: node}
+}
+
+// For cert, config, operator policies.
+// This function returns `annotations`, `isExternal` for `source`,
+// and `severity`, `compliant`, `remediationAction`, `parentPolicyNs`, `parentPolicyName`
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	annos := c.GetAnnotations()
+	// This annotation is large and useless
+	delete(annos, "kubectl.kubernetes.io/last-applied-configuration")
+
+	node.Properties["annotations"] = annos
+
+	node.Properties["isExternal"] = false
+
+	for _, m := range c.GetManagedFields() {
+		if m.Manager == "multicluster-operators-subscription" ||
+			strings.Contains(m.Manager, "argocd") {
+			node.Properties["isExternal"] = true
+		}
+	}
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	policyNsName, ok := c.GetLabels()["policy.open-cluster-management.io/policy"]
+	if ok {
+		stringSlice := strings.Split(policyNsName, ".")
+		// If len(stringSlice) != 2, failed to parse policyNsName
+		if len(stringSlice) == 2 {
+			node.Properties["parentPolicyNs"] = stringSlice[0]
+			node.Properties["parentPolicyName"] = stringSlice[1]
```

Comment:
The parentPolicy should be represented with an edge (or relationship). This is how search "connects" resources.

Reply:
Good point.

@yiraeChristineKim is this easy information to get in the search query to populated the table?

Reply:
@jlpadilla We don't use this for search-page. In Governance, we are building new tabs for configuration/operator/cert policies so This properties just for convenience to search `kind: Policy`

Reply:
@jlpadilla Correct me if I am wrong @mprahl  As I know, to get this info we need to use other API. I tried to avoid that. Configuration, operator, and certificate policies already have the parent policy's namespace and name in their labels. I included this to determine whether it's standalone or part of a Policy that contains configuration, operator, or certificate policies as policy-templates.

Reply:
@yiraeChristineKim could we just leverage the labels in the UI instead since they are already present?

Reply:
@mprahl  doable

Reply:
@yiraeChristineKim okay, let's try with that and not index these additional properties.

Reply:
👍 Good point about not duplicating the information in the label.

FYI, it is the same API to get relationships. You'll just need to update the graphQL query to add `related`.